core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala

```diff
@@ -70,6 +71,17 @@ private[ui] class StagePage(parent: StagesTab) extends WebUIPage("stage") {
 
   private val displayPeakExecutionMemory = parent.conf.getBoolean("spark.sql.unsafe.enabled", true)
 
+  private def getLocalitySummaryString(stageData: StageUIData): String = {
+    val localityCounts = new HashMap[String, Long]
+    stageData.taskData.values.foreach( taskUIData => {
```

Comment:
And while picking nits, how about mapping to (key,value) tuples and calling `toMap` at the end instead of making a `Map` Java-style?